server/src/test/java/org/elasticsearch/search/aggregations/metrics/AvgAggregatorTests.java

```diff
@@ -228,31 +228,31 @@ public void testSummationAccuracy() throws IOException {
 
     public void testUnmappedField() throws IOException {
         AvgAggregationBuilder aggregationBuilder = new AvgAggregationBuilder("_name").field("number");
-        testCase(aggregationBuilder, new DocValuesFieldExistsQuery("number"), iw -> {
+        this.<AvgAggregationBuilder, InternalAvg>testCase(aggregationBuilder, new DocValuesFieldExistsQuery("number"), iw -> {
```

Comment:
Do we need the `<` bits?

Reply:
The `<` bits are ugly but required, otherwise compile fails. Hence I suggested this:
https://github.com/elastic/elasticsearch/pull/55826#discussion_r416151331

Reply:
I see what is up. I wonder if it'd be cleaner to return `V` instead of take a `Consumer<V>`. You'd get the implicit cast from assigning it to the result type you wanted. I wonder if that could save us a little ceremony?

Reply:
For every agg test I have created a local private method named `testAggregation()`. This method converts generic arguments to specific arguments and calls `AggregateTestCase.testCase()`. I think this is a lot cleaner and simpler.

Please check commit 23917bd and let me know what you think.